node/src/types/sync_leap.rs

```diff
@@ -290,20 +295,1676 @@ impl FetchItem for SyncLeap {
+        let lowest_blocks: Vec<_> = sync_leap
+            .trusted_ancestor_headers
+            .iter()
+            .take(2)
+            .cloned()
+            .collect();
+        let middle_blocks: Vec<_> = sync_leap
+            .signed_block_headers
+            .iter()
+            .take(2)
+            .cloned()
+            .map(|block_header_with_metadata| block_header_with_metadata.block_header)
+            .collect();
+
+        let highest_block_height = highest_block.block_header.height();
+        let highest_block_hash = highest_block.block_header.block_hash();
+
+        let sync_leap = SyncLeap {
+            trusted_ancestor_only: false,
+            trusted_block_header: lowest_blocks.first().unwrap().clone(),
+            trusted_ancestor_headers: middle_blocks,
+            signed_block_headers: vec![highest_block.clone()],
+        };
+        assert_eq!(
+            sync_leap
+                .highest_block_header_and_signatures()
+                .0
+                .block_hash(),
+            highest_block.block_header.block_hash()
+        );
+        assert_eq!(sync_leap.highest_block_hash(), highest_block_hash);
+        assert_eq!(sync_leap.highest_block_height(), highest_block_height);
+    }
+
+    #[test]
+    fn should_return_sigs_when_highest_block_is_signed() {
+        // Chain
+        // 0   1   2   3   4   5   6   7   8   9   10   11
+        // S           S           S           S
+        let switch_blocks = [0, 3, 6, 9];
+
+        let mut rng = TestRng::new();
+
+        let query = 5;
+        let trusted_ancestor_headers = [4, 3];
+        let signed_block_headers = [6, 9, 11];
+        let add_proofs = true;
+        let sync_leap = make_test_sync_leap(
+            &mut rng,
+            &switch_blocks,
+            query,
+            &trusted_ancestor_headers,
+            &signed_block_headers,
+            add_proofs,
+        );
+
+        // `sync_leap` is a well formed SyncLeap structure for the test chain. We can use the blocks
+        // it contains to generate SyncLeap structures as required for the test, because we know the
+        // heights of the blocks in the test chain as well as their sigs.
+        let highest_block = sync_leap.signed_block_headers.last().unwrap().clone();
+        let lowest_blocks: Vec<_> = sync_leap
+            .trusted_ancestor_headers
+            .iter()
+            .take(2)
+            .cloned()
+            .collect();
+        let middle_blocks: Vec<_> = sync_leap
+            .signed_block_headers
+            .iter()
+            .take(2)
+            .cloned()
+            .map(|block_header_with_metadata| block_header_with_metadata.block_header)
+            .collect();
+        let sync_leap = SyncLeap {
+            trusted_ancestor_only: false,
+            trusted_block_header: lowest_blocks.first().unwrap().clone(),
+            trusted_ancestor_headers: middle_blocks,
+            signed_block_headers: vec![highest_block],
+        };
```

Comment:
The initial `sync_leap` (which has the benefit of being valid) will suffice here won't it?  These lines could be removed I think.